### PR TITLE
Created new note creation form page with validations.

### LINKED ISSUE #3: Create new note UI

### PR DESCRIPTION
**Pull request by allenjiji**
Fixes #3 
image
image
image

@yedhink _a
Please review this PR.

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx [hunk 1]
```diff
@@ -42,18 +47,38 @@ const NoteForm = ({ onClose, refetch, note, isEdit }) => {
...
-            <Textarea
+            <Input
               required
               className="w-full flex-grow-0"
               label="Description"
               name="description"
-              rows={8}
+            />
+            <Select
+              defaultMenuIsClose
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx:57]: Please share the docs regarding this prop. Also, what happens if we don't provide this prop?

File: app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx [hunk 2]
```diff
@@ -42,18 +47,38 @@ const NoteForm = ({ onClose, refetch, note, isEdit }) => {
...
+              isMulti
+              required
+              className="w-full flex-grow-0"
+              label="Tags"
+              name="tags"
+              options={TAGS_DROPDOWN_DATA}
+              placeholder="Select Role"
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx:74]: The label says `Tags`. But the placeholder says `Role`. Pay attention to the keywords.

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 3]
```diff
@@ -3,11 +3,29 @@ import * as yup from "yup";
...
+    .required("Assigned contact is required"),
+  tags: yup
+    .array()
+    .of(
+      yup.object({
+        label: yup.string().required(),
+        value: yup.number().required(),
+      })
+    )
+    .min(1, "Tags are required"),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js:28]: How many tags are required? Mention it.

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 4]
```diff
@@ -3,11 +3,29 @@ import * as yup from "yup";
...
+  tags: [],
 };
 
 export const NOTES_FORM_VALIDATION_SCHEMA = yup.object().shape({
   title: yup.string().required("Title is required"),
   description: yup.string().required("Description is required"),
+  assignedContacts: yup
+    .object({
+      label: yup.string().required(),
+      value: yup.number().required(),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js:16]: Go through the Slack thread once again. I have mentioned about `oneOf`.

**Review by yedhink — COMMENTED** [review]
@allenjiji

File: app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx (hunk 1 above)
Comment (allenjiji) [reply app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx:57]: Oops! I've no idea where did I find this prop. 
I realize that it's not a valid prop and it makes no effect.

File: app/javascript/src/components/Dashboard/Notes/constants.js (hunk 4 above)
Comment (allenjiji) [reply app/javascript/src/components/Dashboard/Notes/constants.js:16]: I've gone through the thread and made changes as below:
```
assignedContact: yup
    .object({
      label: yup
        .string()
        .oneOf(CONTACTS_DROPDOWN_DATA.map(contact => contact.label)),
      value: yup
        .string()
        .oneOf(CONTACTS_DROPDOWN_DATA.map(contact => contact.value)),
    })
    .nullable()
    .required("Assigned contact is required"),
```
where, 
```
CONTACTS_DROPDOWN_DATA = [
  {
    label: "Ronalds Richards",
    value: "ronald-1",
  },
  {
    label: "Jacob Jones",
    value: "jacob-1",
  }
]
```
Is this the only desired way of using `oneOf` validator in this scenario?
OR
Do we have any option like this? (below code is not correct, but hope it conveys the idea- A way of validating the assignedContact as a whole against the list of maps containing the label and value),
```
assignedContact: yup
    .mixed()
    .oneOf(CONTACTS_DROPDOWN_DATA)
```
Can you please provide an idea of what is `mixed()`?
@yedhink

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Notes/constants.js:16]: >Is this the only desired way of using oneOf validator in this scenario?

@allenjiji The code you've shown is correct. But I prefer to use `object` with `shape` to keep things explicit.

>Can you please provide an idea of what is mixed()?

I have to research on this. Will get back to you when I am free.

**Pull request merged**